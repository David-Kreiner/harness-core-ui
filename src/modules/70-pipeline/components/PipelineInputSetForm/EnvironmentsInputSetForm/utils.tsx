/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get, isEmpty } from 'lodash-es'

import { RUNTIME_INPUT_VALUE, SelectOption } from '@harness/uicore'

import { DeploymentStageConfig } from 'services/cd-ng'
import { StageElementWrapperConfig } from 'services/pipeline-ng'

import { isValueRuntimeInput } from '@common/utils/utils'

import { StageType } from '@pipeline/utils/stageHelpers'
import { DeploymentStageElementConfig } from '@pipeline/utils/pipelineTypes'

export interface PropagateSelectOption extends SelectOption {
  infraLabel?: string
}

export function getPropagateStageOptions(
  stages: StageElementWrapperConfig[],
  stageIndex: number,
  templateTypes: {
    [key: string]: string
  }
): PropagateSelectOption[] {
  if (stages.length && stageIndex > 0) {
    const getStagesAllowedforPropagate = (stageItem: StageElementWrapperConfig): boolean => {
      if (stageItem.stage?.template) {
        const isAlreadyPropagatingValue = !!(stageItem?.stage?.template?.templateInputs?.spec as DeploymentStageConfig)
          ?.environment?.useFromStage?.stage
        const stageType = get(templateTypes, stageItem.stage.template.templateRef)
        return stageType === StageType.DEPLOY && !isAlreadyPropagatingValue
      } else {
        const isAlreadyPropagatingValue = !!(stageItem?.stage?.spec as DeploymentStageConfig)?.environment?.useFromStage
          ?.stage
        const stageType = stageItem?.stage?.type
        const isSingleEnvEmpty = isEmpty(
          (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.environmentRef
        )
        return stageType === StageType.DEPLOY && !isSingleEnvEmpty && !isAlreadyPropagatingValue
      }
    }

    const stageWithEnvV2 = stages.slice(0, stageIndex).filter(getStagesAllowedforPropagate)
    return stageWithEnvV2.map(stageItem => {
      if (stageItem.stage?.template) {
        return {
          label: `Stage [${stageItem.stage?.name}] - Template [${stageItem.stage.template.templateRef}]`,
          value: stageItem.stage?.identifier as string,
          infraLabel: `Stage [${stageItem.stage?.name}] - Template [${stageItem.stage.template.templateRef}]`
        }
      } else {
        const singleEnvironmentRef = (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment
          ?.environmentRef
        const environmentLabel = `Environment [${singleEnvironmentRef}]`

        const singleInfrastructureRef = isValueRuntimeInput(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.infrastructureDefinitions as any
        )
          ? RUNTIME_INPUT_VALUE
          : (stageItem.stage as DeploymentStageElementConfig)?.spec?.environment?.infrastructureDefinitions?.[0]
              ?.identifier
        const infrastructureLabel = `Infrastructure [${singleInfrastructureRef}]`

        return {
          label: `Stage [${stageItem.stage?.name}] - ${environmentLabel}`,
          value: stageItem.stage?.identifier as string,
          infraLabel: `Stage [${stageItem.stage?.name}] - ${infrastructureLabel}`
        }
      }
    })
  }
  return []
}
