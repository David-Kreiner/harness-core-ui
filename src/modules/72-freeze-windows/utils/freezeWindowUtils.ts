/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FreezeDetailedResponse } from 'services/cd-ng'

export const RECURRENCE = ['Daily', 'Weekly', 'Monthly', 'Annually'] as const
export const DOES_NOT_REPEAT = 'Does not repeat'

type Scope = Exclude<FreezeDetailedResponse['freezeScope'], 'unknown' | undefined>

export const scopeText: Record<Scope, string> = {
  account: 'Account',
  org: 'Organization',
  project: 'Project'
}
