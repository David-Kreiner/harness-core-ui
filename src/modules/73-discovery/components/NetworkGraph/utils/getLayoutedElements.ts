/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import ELK, { ElkExtendedEdge, ElkNode } from 'elkjs/lib/elk.bundled'
import type { Edge } from 'reactflow'
import type { LayoutedGraph } from '../types'
import { NodeTypes } from '../constants'

const elk = new ELK()

const options = {
  'elk.algorithm': 'stress'
}

export async function getLayoutedElements({ nodes, edges }: LayoutedGraph): Promise<LayoutedGraph> {
  // Generate elk graph node from reactflow node structure
  // Get the group nodes
  const groupNodes = nodes.filter(node => node.type === NodeTypes.Group)

  // Create parent elk nodes with it's children and edges
  const parentNodes = groupNodes.map(groupNode => {
    return {
      ...groupNode,
      children: nodes
        .filter(node => node.parentNode === groupNode.id)
        .map(node => ({
          ...node,
          // Adjust the target and source handle positions based on the layout
          // direction.
          targetPosition: 'top',
          sourcePosition: 'bottom',

          // Hardcode a width and height for elk to use when layouting.
          width: 100,
          height: 100
        })),
      edges: edges.filter(
        edge => /* istanbul ignore next */ edge.data?.parentNode === groupNode.id
      ) as unknown as ElkExtendedEdge[]
    }
  })

  const graph: ElkNode = {
    id: 'root',
    layoutOptions: options,
    children: parentNodes.map(node => ({
      ...node,
      // Hardcode a width and height for elk to use when layouting.
      width: 0,
      height: 0
    })),
    edges: edges as unknown as ElkExtendedEdge[]
  }

  const elkGraph = await elk.layout(graph)

  /* istanbul ignore next */
  const flattenedNodes = elkGraph?.children?.map(node => ({
    ...node,
    // React Flow expects a position property on the node instead of `x`
    // and `y` fields.
    position: { x: node.x, y: node.y }
  }))

  /* istanbul ignore next */
  elkGraph?.children?.map(node => {
    if (node.children) {
      /* istanbul ignore next */
      flattenedNodes?.push(
        ...node.children.map(n => ({
          ...n,
          // React Flow expects a position property on the node instead of `x`
          // and `y` fields.
          position: { x: n.x, y: n.y }
        }))
      )
    }
  })

  const layoutedGraph = {
    nodes: flattenedNodes,
    edges: elkGraph.edges as unknown as Edge<unknown>[]
  } as LayoutedGraph

  return layoutedGraph
}
