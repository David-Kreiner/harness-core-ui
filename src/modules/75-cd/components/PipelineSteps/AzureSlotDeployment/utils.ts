import { defaultTo } from 'lodash-es'
import { RUNTIME_INPUT_VALUE, MultiTypeInputType } from '@harness/uicore'
import type { StageElementWrapper } from '@pipeline/utils/pipelineTypes'
import type { StageElementConfig, StageElementWrapperConfig, PipelineInfoConfig } from 'services/pipeline-ng'
import type { InfraStructureDefinitionYaml, DeploymentStageConfig, EnvironmentYamlV2 } from 'services/cd-ng'

interface EnvironmentYamlV2Runtime extends Omit<EnvironmentYamlV2, 'infrastructureDefinitions'> {
  infrastructureDefinitions?: InfraStructureDefinitionYaml[] | string
}

interface DeploymentStageConfigAzureWebName extends Omit<DeploymentStageConfig, 'environment'> {
  environment: EnvironmentYamlV2Runtime
}

export interface DeploymentStageElementConfigAzureWebName extends Omit<StageElementConfig, 'spec'> {
  spec?: DeploymentStageConfigAzureWebName
}
export type SelectedStageType = StageElementWrapper<DeploymentStageElementConfigAzureWebName>

export function getStage(stageId: string, pipeline?: PipelineInfoConfig): StageElementWrapperConfig | undefined {
  if (pipeline?.stages) {
    let responseStage: StageElementWrapperConfig | undefined = undefined
    pipeline.stages.forEach(item => {
      if (item.stage && item.stage.identifier === stageId) {
        responseStage = item
      } else if (item.parallel) {
        return item.parallel.forEach(node => {
          if (node.stage?.identifier === stageId) {
            responseStage = node
          }
        })
      }
    })
    return responseStage
  }
  return
}

export const isRuntimeEnvId = (selectedStage: SelectedStageType): boolean => {
  const envId = selectedStage?.stage?.spec?.environment?.environmentRef
  return !!(envId && envId === RUNTIME_INPUT_VALUE)
}

export const isRuntimeInfraId = (selectedStage: SelectedStageType): boolean => {
  const infraId = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions
  return infraId === RUNTIME_INPUT_VALUE
}

export const isMultiEnv = (selectedStage: SelectedStageType): boolean => {
  const isEnvs = !!selectedStage?.stage?.spec?.environments || !!selectedStage?.stage?.spec?.environmentGroup
  return !!isEnvs
}

export const getEnvId = (selectedStage: SelectedStageType): string => {
  if (isMultiEnv(selectedStage || isRuntimeEnvId(selectedStage))) {
    return ''
  }

  return defaultTo(selectedStage?.stage?.spec?.environment?.environmentRef, '')
}

export const getInfraId = (selectedStage: SelectedStageType): string => {
  const envId = getEnvId(selectedStage)
  if (!envId || isMultiEnv(selectedStage)) {
    return ''
  }

  if (!isRuntimeEnvId(selectedStage) && !isRuntimeInfraId(selectedStage)) {
    const [infra] = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return infra.identifier
  }

  return ''
}
/* eslint-disable */
export const getEnvIdRuntime = (stageId: string, values: any): string => {
  const stage = getStage(stageId, values) as SelectedStageType
  const envId = stage?.stage?.spec?.environment?.environmentRef
  return envId || ''
}
/* eslint-disable */
export const getInfraIdRuntime = (stageId: string, values: any): string => {
  const stage = getStage(stageId, values) as SelectedStageType

  if (stage?.stage?.spec?.environment?.infrastructureDefinitions?.[0]) {
    const [{ identifier }] = stage?.stage?.spec?.environment
      ?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return identifier
  }
  return ''
}

export const getAllowableTypes = (selectedStage: SelectedStageType): MultiTypeInputType[] => {
  if (isMultiEnv(selectedStage)) {
    return [MultiTypeInputType.EXPRESSION]
  }

  return [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]
}
