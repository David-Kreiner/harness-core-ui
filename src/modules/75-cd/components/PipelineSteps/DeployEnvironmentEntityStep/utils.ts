/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { getMultiTypeFromValue, MultiTypeInputType, RUNTIME_INPUT_VALUE } from '@harness/uicore'
import { defaultTo, get, isEmpty, isNil, set } from 'lodash-es'
import type { EnvironmentYamlV2 } from 'services/cd-ng'
import type { DeployEnvironmentEntityConfig, DeployEnvironmentEntityFormState } from './types'

export function processInitialValues(initialValues: DeployEnvironmentEntityConfig): DeployEnvironmentEntityFormState {
  const formState = {}

  if (initialValues.environment && initialValues.environment.environmentRef) {
    if (getMultiTypeFromValue(initialValues.environment.environmentRef) === MultiTypeInputType.RUNTIME) {
      return {
        environment: RUNTIME_INPUT_VALUE
      }
    }

    const infrastructureDefinition = initialValues.environment.infrastructureDefinitions?.[0]
    const isInfrastructureRuntime =
      (initialValues.environment.infrastructureDefinitions as unknown as string) === RUNTIME_INPUT_VALUE

    return {
      environment: initialValues.environment.environmentRef,
      environmentInputs:
        getMultiTypeFromValue(initialValues.environment.environmentRef) === MultiTypeInputType.FIXED
          ? { [initialValues.environment.environmentRef]: initialValues.environment?.environmentInputs }
          : {},
      infrastructure: isInfrastructureRuntime ? RUNTIME_INPUT_VALUE : infrastructureDefinition?.identifier,
      // ? Can there be a better condition to identify?
      ...(!!infrastructureDefinition?.identifier && {
        infrastructureInputs: !isEmpty(infrastructureDefinition?.identifier)
          ? {
              [initialValues.environment.environmentRef]: {
                [infrastructureDefinition.identifier]: infrastructureDefinition?.inputs
              }
            }
          : {}
      })
    }
  } else if (initialValues.environments) {
    if (getMultiTypeFromValue(initialValues.environments?.values as unknown as string) !== MultiTypeInputType.FIXED) {
      set(formState, 'environments', initialValues.environments?.values)
      set(formState, 'environmentInputs', {})
      set(formState, 'infrastructures', [])
      set(formState, 'infrastructureInputs', {})
    } else {
      defaultTo(initialValues.environments?.values, []).map((environment: EnvironmentYamlV2, index: number) => {
        set(formState, `environments.${index}`, {
          label: environment.environmentRef,
          value: environment.environmentRef
        })
        set(formState, `environmentInputs.${environment.environmentRef}`, environment.environmentInputs)
        set(formState, `serviceOverrideInputs.${environment.environmentRef}`, environment.serviceOverrideInputs)
        if (Array.isArray(environment.infrastructureDefinitions)) {
          environment.infrastructureDefinitions.map((infrastructure, infrastructureIndex) => {
            set(formState, `infrastructures.${environment.environmentRef}.${infrastructureIndex}`, {
              label: infrastructure.identifier,
              value: infrastructure.identifier
            })
            set(
              formState,
              `infrastructureInputs.${environment.environmentRef}.${infrastructure.identifier}`,
              infrastructure.inputs
            )
          })
        } else {
          set(formState, `infrastructures.${environment.environmentRef}`, environment.infrastructureDefinitions)
        }
      })
    }
    set(formState, `parallel`, defaultTo(initialValues.environments?.metadata?.parallel, true))
  } else if (initialValues.environmentGroup && initialValues.environmentGroup.envGroupRef) {
    set(formState, `environmentGroup`, initialValues.environmentGroup.envGroupRef)

    if (
      getMultiTypeFromValue(initialValues.environmentGroup?.envGroupRef as unknown as string) ===
      MultiTypeInputType.FIXED
    ) {
      if (
        getMultiTypeFromValue(initialValues.environmentGroup?.environments as unknown as string) !==
        MultiTypeInputType.FIXED
      ) {
        set(formState, 'environments', initialValues.environmentGroup.environments)
        set(formState, 'environmentInputs', {})
        set(formState, 'infrastructures', [])
        set(formState, 'infrastructureInputs', {})
      } else {
        // This section is same as multi env. Move to function
        defaultTo(initialValues.environmentGroup.environments, []).map(
          (environment: EnvironmentYamlV2, index: number) => {
            set(formState, `environments.${index}`, {
              label: environment.environmentRef,
              value: environment.environmentRef
            })
            set(formState, `environmentInputs.${environment.environmentRef}`, environment.environmentInputs)
            set(formState, `serviceOverrideInputs.${environment.environmentRef}`, environment.serviceOverrideInputs)
            if (Array.isArray(environment.infrastructureDefinitions)) {
              environment.infrastructureDefinitions.map((infrastructure, infrastructureIndex) => {
                set(formState, `infrastructures.${environment.environmentRef}.${infrastructureIndex}`, {
                  label: infrastructure.identifier,
                  value: infrastructure.identifier
                })
                set(
                  formState,
                  `infrastructureInputs.${environment.environmentRef}.${infrastructure.identifier}`,
                  infrastructure.inputs
                )
              })
            } else {
              set(formState, `infrastructures.${environment.environmentRef}`, environment.infrastructureDefinitions)
            }
          }
        )
      }
    }
  }

  return formState
}

// TODO: Add type check to optionally extends keys
export type ChangeTypeOfKeys<T, N, K extends keyof T> = Pick<T, K> &
  {
    [key in keyof T]: T[key] | N
  }

export function processFormValues(data: DeployEnvironmentEntityFormState): DeployEnvironmentEntityConfig {
  if (!isNil(data.environment)) {
    // ! Do not merge this with the other returns even if they look similar. It makes it confusing to read
    if (getMultiTypeFromValue(data.environment) === MultiTypeInputType.RUNTIME) {
      return {
        environment: {
          environmentRef: RUNTIME_INPUT_VALUE,
          deployToAll: false,
          environmentInputs: RUNTIME_INPUT_VALUE as any,
          serviceOverrideInputs: RUNTIME_INPUT_VALUE as any,
          infrastructureDefinitions: RUNTIME_INPUT_VALUE as any
        }
      }
    }

    return {
      environment: {
        environmentRef: data.environment,
        ...(!!data.environmentInputs?.[data.environment] && {
          environmentInputs: data.environmentInputs[data.environment]
        }),
        deployToAll: false,
        ...(data.environment &&
          !!data.infrastructure && {
            infrastructureDefinitions:
              getMultiTypeFromValue(data.infrastructure) === MultiTypeInputType.RUNTIME
                ? (data.infrastructure as any)
                : [
                    {
                      identifier: data.infrastructure,
                      inputs: get(data, `infrastructureInputs.${data.environment}.${data.infrastructure}`)
                    }
                  ]
          })
      }
    }
  } else if (!isNil(data.environmentGroup)) {
    if (getMultiTypeFromValue(data.environmentGroup) === MultiTypeInputType.RUNTIME) {
      return {
        environmentGroup: {
          envGroupRef: RUNTIME_INPUT_VALUE,
          deployToAll: RUNTIME_INPUT_VALUE as any,
          environments: RUNTIME_INPUT_VALUE as any
        }
      }
    }

    return {
      environmentGroup: {
        envGroupRef: data.environmentGroup,
        deployToAll:
          getMultiTypeFromValue(data.environments) === MultiTypeInputType.RUNTIME
            ? (RUNTIME_INPUT_VALUE as any)
            : isEmpty(data.environments),
        environments: Array.isArray(data.environments)
          ? data.environments?.map(environment => {
              const selectedInfrastructures = data.infrastructures?.[environment.value as string]
              const isInfrastructureRuntime = (selectedInfrastructures as unknown as string) === RUNTIME_INPUT_VALUE

              return {
                environmentRef: environment.value as string,
                ...(!!data.environmentInputs?.[environment.value as string] && {
                  environmentInputs: data.environmentInputs[environment.value as string]
                }),
                deployToAll: isInfrastructureRuntime ? (RUNTIME_INPUT_VALUE as any) : !selectedInfrastructures,
                ...(!!selectedInfrastructures && {
                  infrastructureDefinitions: Array.isArray(selectedInfrastructures)
                    ? selectedInfrastructures.map(infrastructure => ({
                        identifier: infrastructure.value as string,
                        inputs: data.infrastructureInputs?.[environment.value as string][infrastructure.value as string]
                      }))
                    : selectedInfrastructures
                })
              }
            })
          : data.environments
      }
    }
  } else if (!isNil(data.environments)) {
    return {
      environments: {
        metadata: {
          parallel: data.parallel
        },
        values: Array.isArray(data.environments)
          ? data.environments?.map(environment => {
              const selectedInfrastructures = data.infrastructures?.[environment.value as string]
              const isInfrastructureRuntime = (selectedInfrastructures as unknown as string) === RUNTIME_INPUT_VALUE

              return {
                environmentRef: environment.value as string,
                ...(!!data.environmentInputs?.[environment.value as string] && {
                  environmentInputs: data.environmentInputs[environment.value as string]
                }),
                deployToAll: isInfrastructureRuntime ? (RUNTIME_INPUT_VALUE as any) : !selectedInfrastructures,
                ...(!!selectedInfrastructures && {
                  infrastructureDefinitions: Array.isArray(selectedInfrastructures)
                    ? selectedInfrastructures.map(infrastructure => ({
                        identifier: infrastructure.value as string,
                        inputs: data.infrastructureInputs?.[environment.value as string][infrastructure.value as string]
                      }))
                    : selectedInfrastructures
                })
              }
            })
          : data.environments
      }
    }
  }
  return {}
}
