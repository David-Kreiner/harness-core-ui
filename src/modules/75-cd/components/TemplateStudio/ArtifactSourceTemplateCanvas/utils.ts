import type { ArtifactType } from '@pipeline/components/ArtifactsSelection/ArtifactInterface'
import { ArtifactToConnectorMap } from '@pipeline/components/ArtifactsSelection/ArtifactHelper'

export interface Params {
  oldConnectorRef?: string
  newConnectorRef?: string
  artifactType: ArtifactType
}

export const isConnectorRefUpdated = (params: Params) => {
  const { oldConnectorRef, newConnectorRef, artifactType } = params
  const isConnectorRefApplicable = Boolean(ArtifactToConnectorMap[artifactType])

  return Boolean(isConnectorRefApplicable && oldConnectorRef && newConnectorRef && oldConnectorRef !== newConnectorRef)
}
