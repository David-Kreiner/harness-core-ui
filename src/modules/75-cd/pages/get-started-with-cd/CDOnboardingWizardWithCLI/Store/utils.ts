export const parseJSON = (str: string): object => {
  try {
    return JSON.parse(str)
  } catch (e) {
    return {}
  }
}
