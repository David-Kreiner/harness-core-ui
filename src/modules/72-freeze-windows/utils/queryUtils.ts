import type { SelectOption } from '@harness/uicore'
import { queryParamDecodeAll } from '@common/hooks/useQueryParams'
import type { FreezeListUrlQueryParams } from '@freeze-windows/types'
import { DEFAULT_PAGE_INDEX, DEFAULT_PAGE_SIZE, DEFAULT_PIPELINE_LIST_TABLE_SORT } from '@pipeline/utils/constants'

export const getQueryParamOptions = () => ({
  parseArrays: true,
  decoder: queryParamDecodeAll(),
  processQueryParams(params: FreezeListUrlQueryParams): FreezeListUrlQueryParams {
    return {
      ...params,
      page: params.page ?? DEFAULT_PAGE_INDEX,
      size: params.size ?? DEFAULT_PAGE_SIZE,
      sort: params.sort ?? DEFAULT_PIPELINE_LIST_TABLE_SORT
    }
  }
})

export const getDefaultCalenderFilter = (): [Date, Date] => {
  const start = new Date()
  start.setDate(start.getDate() - 7)
  start.setHours(0, 0, 0, 0)

  const end = new Date()
  end.setHours(23, 59, 59, 999)

  return [start, end]
}

export const STATUS_OPTIONS: SelectOption[] = [
  { label: 'Enabled', value: 'Enabled' },
  { label: 'Disabled', value: 'Disabled' }
]
