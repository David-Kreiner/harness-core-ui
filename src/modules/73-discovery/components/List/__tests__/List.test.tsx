/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import List from '../List'

const title = 'What is a network map?'
const description =
  'A map of services designed to ensure that the different entity types within Kubernetes can communicate'

describe('List', () => {
  test('render component paint lists', async () => {
    const { container } = render(
      <TestWrapper>
        <List title={title} content={description} margin={{ top: 'medium', bottom: 'xlarge' }} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('it should have correct color props', async () => {
    const { getByText } = render(
      <TestWrapper>
        <List title={title} content={description} />
      </TestWrapper>
    )

    expect(getByText(title)).toBeInTheDocument()
    expect(getByText(title)).toHaveClass('StyledProps--color-grey900')

    expect(getByText(description)).toBeInTheDocument()
    expect(getByText(description)).toHaveClass('StyledProps--color-grey600')
  })
})
