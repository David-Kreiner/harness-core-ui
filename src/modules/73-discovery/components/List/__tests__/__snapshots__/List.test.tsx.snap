// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render component paint lists 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--margin-bottom-xlarge"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h6 StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey900 StyledProps--margin-bottom-xlarge"
    >
      What is a network map?
    </p>
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-body StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
    >
      A map of services designed to ensure that the different entity types within Kubernetes can communicate
    </p>
  </div>
</div>
`;
